### PR TITLE
Fix documentation issues regarding QBDI limitations

### PR DESCRIPTION
**Pull request by haxelion**
This should explain issues like #27 and #33.

### CODE REVIEW

File: docs/source/intro.rst [hunk 1]
```diff
@@ -60,3 +60,26 @@ process, and then instrumented as instructed by the instrumentation tool. This i
...
 code cache.
+
+.. _intro_limitations:
+
+Limitations
+-----------
+
+The **host** and the **guest** share the same process and thus the same resources. This means that 
+they use the same heap and the same libraries and this will cause issues with any non-reentrant 
+code. We could have chosen to shield users from those issue by forbidding instrumentation tools to
```
Comment (nezetic) [inline docs/source/intro.rst]: those issue**s**

File: docs/source/intro.rst [hunk 2]
```diff
@@ -60,3 +60,26 @@ process, and then instrumented as instructed by the instrumentation tool. This i
...
+-----------
+
+The **host** and the **guest** share the same process and thus the same resources. This means that 
+they use the same heap and the same libraries and this will cause issues with any non-reentrant 
+code. We could have chosen to shield users from those issue by forbidding instrumentation tools to 
+use any external libraries like some other DBI frameworks have done. However we believe this is 
+an overblown issue and that there are effective mechanisms to mitigate the problem. Nonetheless 
+users need to be aware of this design limitation and the mitigations side-effects. 
+
+For example tracing the heap memory allocator will cause deadlocks because it is not reentrant.
```
Comment (nezetic) [inline docs/source/intro.rst]: For example **,**

File: docs/source/intro.rst [hunk 3]
```diff
@@ -60,3 +60,26 @@ process, and then instrumented as instructed by the instrumentation tool. This i
...
+
+The **host** and the **guest** share the same process and thus the same resources. This means that 
+they use the same heap and the same libraries and this will cause issues with any non-reentrant 
+code. We could have chosen to shield users from those issue by forbidding instrumentation tools to 
+use any external libraries like some other DBI frameworks have done. However we believe this is 
+an overblown issue and that there are effective mechanisms to mitigate the problem. Nonetheless 
+users need to be aware of this design limitation and the mitigations side-effects. 
+
+For example tracing the heap memory allocator will cause deadlocks because it is not reentrant. 
+There are others problematic cases but they are mostly limited to the standard C library and the OS
```
Comment (nezetic) [inline docs/source/intro.rst]: There are other

File: docs/source/intro.rst [hunk 4]
```diff
@@ -60,3 +60,26 @@ process, and then instrumented as instructed by the instrumentation tool. This i
...
+The **host** and the **guest** share the same process and thus the same resources. This means that 
+they use the same heap and the same libraries and this will cause issues with any non-reentrant 
+code. We could have chosen to shield users from those issue by forbidding instrumentation tools to 
+use any external libraries like some other DBI frameworks have done. However we believe this is 
+an overblown issue and that there are effective mechanisms to mitigate the problem. Nonetheless 
+users need to be aware of this design limitation and the mitigations side-effects. 
+
+For example tracing the heap memory allocator will cause deadlocks because it is not reentrant. 
+There are others problematic cases but they are mostly limited to the standard C library and the OS 
+loader. To avoid such issues we have an execution brokering system that allow to whitelist/blacklist
```
Comment (nezetic) [inline docs/source/intro.rst]: that allow**s**

File: docs/source/intro.rst [hunk 5]
```diff
@@ -60,3 +60,26 @@ process, and then instrumented as instructed by the instrumentation tool. This i
...
+
+For example tracing the heap memory allocator will cause deadlocks because it is not reentrant. 
+There are others problematic cases but they are mostly limited to the standard C library and the OS 
+loader. To avoid such issues we have an execution brokering system that allow to whitelist/blacklist 
+specific pieces of code. These will be executed outside of the instrumentation process via a call 
+hooking mechanism. This execution broker system is documented in :ref:`execution-filtering-c` and 
+:ref:`execution-filtering`.
+
+Moreover, the **host** relies on the loader loading and initializing its library dependencies which 
+means the instrumentation process can not be started before the loader has finished its job. As a
```
Comment (nezetic) [inline docs/source/intro.rst]: can not -> cannot

**Comment by haxelion** [comment]
Ah yes, grammar :)
It should be good now.

**Pull request merged**